packages/order-utils/src/rate_utils.ts

```diff
@@ -0,0 +1,48 @@
+import { schemas } from '@0xproject/json-schemas';
+import { SignedOrder } from '@0xproject/types';
+import { BigNumber } from '@0xproject/utils';
+
+import { assert } from './assert';
+import { constants } from './constants';
+
+export const rateUtils = {
+    /**
+     * Takes a signed order and calculates the fee adjusted rate (takerAsset/makerAsset) by calculating how much takerAsset
+     * is required to cover the fees (feeRate * takerFee), adding the takerAssetAmount and dividing by makerAssetAmount
+     * @param   signedOrder     An object that conforms to the signedOrder interface
+     * @param   feeRate         The market rate of ZRX denominated in takerAssetAmount
+     *                          (ex. feeRate is 0.1 takerAsset/ZRX if it takes 1 unit of takerAsset to buy 10 ZRX)
+     *                          Defaults to 0
+     * @return  The rate (takerAsset/makerAsset) of the order adjusted for fees
+     */
+    getFeeAdjustedRateOfOrder(signedOrder: SignedOrder, feeRate: BigNumber = constants.ZERO_AMOUNT): BigNumber {
+        assert.doesConformToSchema('signedOrder', signedOrder, schemas.signedOrderSchema);
+        assert.isBigNumber('feeRate', feeRate);
+        assert.assert(
+            feeRate.gte(constants.ZERO_AMOUNT),
+            `Expected feeRate: ${feeRate} to be greater than or equal to 0`,
+        );
+        const takerAssetAmountNeededToPayForFees = signedOrder.takerFee.mul(feeRate);
+        const totalTakerAssetAmount = takerAssetAmountNeededToPayForFees.plus(signedOrder.takerAssetAmount);
+        const rate = totalTakerAssetAmount.div(signedOrder.makerAssetAmount);
+        return rate;
+    },
+    /**
+     * Takes a signed fee order (makerAssetData corresponds to ZRX and takerAssetData corresponds to WETH) and calculates
+     * the fee adjusted rate (WETH/ZRX) by dividing the takerAssetAmount by the makerAmount minus the takerFee
+     * @param   signedFeeOrder    An object that conforms to the signedOrder interface
+     * @return  The rate (WETH/ZRX) of the fee order adjusted for fees
+     */
+    getFeeAdjustedRateOfFeeOrder(signedFeeOrder: SignedOrder): BigNumber {
```

Comment:
Should we assert that the pair is indeed WETH/ZRX? It could be another schema ?

Reply:
It's a bit hard to assert w/o having to pass in the network context, i think its fine, the calculations don't necessarily rely on the order being WETH/ZRX, the results just don't make sense if a different type of order is passed in